src/Compilers/Core/MSBuildTask/TaskCompilerServerLogger.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Build.Utilities;
+using Microsoft.CodeAnalysis.CommandLine;
+
+namespace Microsoft.CodeAnalysis.BuildTasks;
+
+/// <summary>
+/// Logs to both the MSBuild task's output and the inner compiler server logger.
+/// </summary>
+internal sealed class TaskCompilerServerLogger(
+    TaskLoggingHelper taskLogger,
+    ICompilerServerLogger inner)
+    : ICompilerServerLogger
+{
+    private readonly TaskLoggingHelper _taskLogger = taskLogger;
+    private readonly ICompilerServerLogger _inner = inner;
+
+    public bool IsLogging => true;
+
+    public void Log(string message)
+    {
+        _inner.Log(message);
+        _taskLogger.LogMessage(message);
```

Comment:
Do you have an example binlog we can look at with this output? I've definitely thought about doing this a few times. My concern is that the compiler logger was designed for debugging the compiler, there is a lot of outptut I'm not sure would be valuable in the binlog (or make it harder to read).

Reply:
It looks like this:

<img width="2768" height="1342" alt="image" src="https://github.com/user-attachments/assets/ed9695b7-c34c-40b4-b49c-e874eaad5152" />

(except the duplicate message at the end, I've fixed that since taking the screenshot)

Reply:
Also it doesn't feel verbose at all, except the BuildResponseFile line which I can probably remove (it's redundant anyway) from the binlog.

Reply:
Now it looks like this:

<img width="2764" height="633" alt="image" src="https://github.com/user-attachments/assets/517d2560-4231-48fc-a8b9-4a2ffef4a28d" />